yb-voyager/cmd/archiveChangesCommand.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+var archiveChangesCmd = &cobra.Command{
+	Use:   "changes",
+	Short: "This command will archive the streaming data from the source database",
+	Long:  `This command will archive the streaming data from the source database.`,
+
+	PreRun: func(cmd *cobra.Command, args []string) {
+		validateCommonArchiveFlags()
+	},
+
+	Run: archiveChangesCommandFn,
+}
+
+func archiveChangesCommandFn(cmd *cobra.Command, args []string) {
+	var err error
+	metaDB, err = NewMetaDB(exportDir)
+	if err != nil {
+		utils.ErrExit("Failed to initialize meta db: %s", err)
+	}
+
+	moveToChanged := cmd.Flags().Changed("move-to")
+	deleteChanged := cmd.Flags().Changed("delete")
+	if (moveToChanged && !deleteChanged) || (!moveToChanged && deleteChanged) {
+		copier := NewEventSegmentCopier(moveDestination)
+		deleter := NewEventSegmentDeleter(utilizationThreshold)
+		var wg sync.WaitGroup
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := copier.Run()
+			if err != nil {
+				utils.ErrExit("Error while copying segments: %v", err)
+			}
+		}()
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := deleter.Run()
+			if err != nil {
+				utils.ErrExit("Error while deleting segments: %v", err)
+			}
+		}()
+
+		wg.Wait()
```

Comment:
Maybe something for later: we should exit `archiver` if cutover/switchover is COMPLETE and all segments are already archived/deleted.